### PR TITLE
[Refactoring] Make inner classes static

### PR DESCRIPTION
**Pull request by Shadow-Devil**
## Purpose
A non-static inner class has a reference to its outer class, and access to the outer class' fields and methods. That class reference makes the inner class larger and could cause the outer class instance to live in memory longer than necessary.

If the reference to the outer class isn’t used, it is more efficient to make the inner class static (also called nested). If the reference is used only in the class constructor, then explicitly pass a class reference to the constructor. If the inner class is anonymous, it will also be necessary to name it.

However, while a nested/static class would be more efficient, it’s worth noting that there are semantic differences between an inner class and a nested one:

- an inner class can only be instantiated within the context of an instance of the outer class.
- a nested (static) class can be instantiated independently of the outer class.

https://sonarcloud.io/organizations/ballerina-platform/rules?open=java%3AS2694&q=inner+class&tab=why

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Comment by Shadow-Devil** [comment]
Remove state status

File: compiler/ballerina-lang/src/test/java/io/ballerina/projects/FileSystemRepositoryTests.java [hunk 1]
```diff
@@ -39,7 +39,7 @@
  */
 public class FileSystemRepositoryTests {
 
-    class MockFileSystemRepository extends FileSystemRepository {
+    static class MockFileSystemRepository extends FileSystemRepository {
```
Comment (HindujaB) [inline compiler/ballerina-lang/src/test/java/io/ballerina/projects/FileSystemRepositoryTests.java:42]: Shall we make this `private`? I think this is not used outside of the `FileSystemRepositoryTests`. 
We can check the same in other places as well.

**Review by HindujaB — APPROVED** [review]
LGTM

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**